### PR TITLE
Customer chat start

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
@amuramoto CCP post final code

### CODE REVIEW

File: customer-chat/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Quick Start Tutorial - Starter Project
```
Comment (amuramoto) [inline customer-chat/README.md]: Update readme - still calls this the quick start

File: customer-chat/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+This is a starter project you can use to as a starting point for completing the [Messenger Platform quick start tutorial](https://developers.facebook.com/docs/messenger-platform/getting-started/quick-start). 
+
+To download the completed quick start code, download it from the [master branch](https://github.com/fbsamples/messenger-platform-samples/tree/master/quick-start) of this repo.
```
Comment (amuramoto) [inline customer-chat/README.md]: update link to starter once available

File: customer-chat/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Quick Start Tutorial - Starter Project
+
+This is a starter project you can use to as a starting point for completing the [Messenger Platform quick start tutorial](https://developers.facebook.com/docs/messenger-platform/getting-started/quick-start).
```
Comment (amuramoto) [inline customer-chat/README.md]: update link

File: customer-chat/app.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+'use strict';
+
+// Imports dependencies and set up http server
+const
+    request = require('request'),
+    express = require('express'),
+    body_parser = require('body-parser'),
+    app = express().use(body_parser.json()); // creates express http server
+app.use(express.static('public'));
```
Comment (amuramoto) [inline customer-chat/app.js]: add a code comment to say what this is for

File: customer-chat/app.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+// TODO: Add values here
+const access_token = '';
```
Comment (amuramoto) [inline customer-chat/app.js]: group your consts with the others at the top before the app.use

File: customer-chat/app.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+app.listen(process.env.PORT || 5000, () => console.log('webhook is listening'));
+
+// receives webhook events from Messenger Platform
+app.post('/webhook', (req, res) => {
+
+    // parse messaging array
+    const webhook_events = req.body.entry[0];
+
+    // Secondary Receiver is in control - listen on standby channel
+    if (webhook_events.standby) {
```
Comment (amuramoto) [inline customer-chat/app.js]: I don't think standby is used in this tutorial?

File: customer-chat/app.js [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+        // iterate webhook events from standby channel
+        webhook_events.standby.forEach(event => {
+            handleEvent(event.sender.id, event);
+        });
+    }
+
+    // Bot is in control - listen for messages
+    if (webhook_events.messaging) {
```
Comment (amuramoto) [inline customer-chat/app.js]: if standby isn't being used then this if statement probably isn't necessary

File: customer-chat/app.js [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+    let request_body = {
+        recipient: {
+            id: sender_psid
+        },
+        target_app_id
+    };
+
+    // Send the HTTP request to the Messenger Platform
+    request({
+        "uri": "https://graph.facebook.com/v2.6/me/pass_thread_control",
```
Comment (amuramoto) [inline customer-chat/app.js]: let's remove the version from the path

File: customer-chat/app.js [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+    let request_body = {
+        "recipient": {
+            "id": sender_psid
+        },
+        "message": response
+    };
+
+    // Send the HTTP request to the Messenger Platform
+    request({
+        "uri": "https://graph.facebook.com/v2.6/me/messages",
```
Comment (amuramoto) [inline customer-chat/app.js]: let's remove the version from the path

File: customer-chat/package.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "messenger-quick-start",
+  "version": "1.0.0",
+  "description": "An example webhook for Facebook Messenger quick start tutorial.",
```
Comment (amuramoto) [inline customer-chat/package.json]: Update description

File: customer-chat/public/chat.html [hunk 11]
```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
```
Comment (amuramoto) [inline customer-chat/public/chat.html]: Fix formatting. None of the nested tags are indented

Comment (amuramoto) [reply customer-chat/public/chat.html]: Also needs to be fixed in the blog post

File: customer-chat/app.js (hunk 9 above)
Comment (ChrisChinchilla) [reply customer-chat/app.js]: @amuramoto I've never seen an example without the version in the path, can you just remove it and it uses latest?

File: customer-chat/app.js [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+    res.sendStatus(200);
+
+});
+
+
+// Accepts GET requests at the /webhook endpoint
+app.get('/webhook', (req, res) => {
+
+    /** UPDATE YOUR VERIFY TOKEN **/
+    const VERIFY_TOKEN = "TOKEN";
```
[the remark this answers is not in the document]
Comment (ChrisChinchilla) [reply customer-chat/app.js]: @amuramoto OK, I think this it's this way in all other tutorials and places in the docs, so might be worth checking…

File: customer-chat/app.js (hunk 7 above)
Comment (ChrisChinchilla) [reply customer-chat/app.js]: @amuramoto It is used.

File: customer-chat/app.js (hunk 5 above)
Comment (ChrisChinchilla) [reply customer-chat/app.js]: @amuramoto OK, but again it's like this in the docs and other tutorials, so might be worth checking.

Comment (ChrisChinchilla) [reply customer-chat/app.js]: Maybe 🤔

File: customer-chat/README.md (hunk 3 above)
Comment (ChrisChinchilla) [reply customer-chat/README.md]: @amuramoto Done, when this review is finished I'll create the starter project and then change again.

File: customer-chat/app.js (hunk 8 above)
Comment (amuramoto) [reply customer-chat/app.js]: @ChrisChinchilla

File: customer-chat/app.js (hunk 9 above)
Comment (amuramoto) [reply customer-chat/app.js]: @ChrisChinchilla  yup

Comment (amuramoto) [reply customer-chat/app.js]: Actually right now it uses 2.11 I think, but normally it uses latest

**Pull request merged**